redash/models/__init__.py

```diff
@@ -690,6 +690,20 @@ def parameters(self):
     def parameterized(self):
         return ParameterizedQuery(self.query_text, self.parameters)
 
+    @property
+    def dashboard_api_keys(self):
+        query = """SELECT api_keys.api_key
+                   FROM api_keys
+                   JOIN dashboards ON object_id = dashboards.id
+                   JOIN widgets ON dashboards.id = widgets.dashboard_id
+                   JOIN visualizations ON widgets.visualization_id = visualizations.id
+                   JOIN queries ON visualizations.query_id = queries.id
+                   WHERE object_type='dashboards'
+                     AND queries.api_key = :api_key"""
+
+        api_keys = db.session.execute(query, {'api_key': self.api_key}).fetchall()
```

Comment:
Why do you filter the queries with `api_key` and not `id`? 🤔 

If you use `id` you can even skip the join with the queries table and just look up by `visualizations.query_id = :id`.

Reply:
Also, when thinking of it it's strange that you ask a _query_ for API keys of _dashboards_.

It probably makes more sense to ask the dashboard for all the query ids it holds. But I can see how it might complicate the code, so we can skip this for now.

Reply:
> Why do you filter the queries with api_key and not id? 🤔

![cfni](https://user-images.githubusercontent.com/289488/56092893-d61adc00-5eca-11e9-83de-7df3ca0a4d3e.gif)

must be a refactoring leftover or something 🤦‍♂

Reply:
f8380301